### PR TITLE
[WB-9964] Support for multiple artifact filesets in mock_server

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-9964](https://wandb.atlassian.net/browse/WB-9964)
Fixes #NNNN

Description
-----------

Adds support so the mock_server can support several artifact file sets and retrieve them separately.

Associated yea-wandb PR: https://github.com/wandb/yea-wandb/pull/69

Testing
-------
How was this PR tested?

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: tests/utils/mock_server.py [hunk 1]
```diff
@@ -1184,6 +1193,7 @@ def graphql():
         if "mutation UseArtifact(" in body["query"]:
             used_name = body.get("variables", {}).get("usedAs", None)
             ctx["used_artifact_info"] = {"used_name": used_name}
+            # ctx[]
```
Comment (kptkin) [inline tests/utils/mock_server.py:1196]: remove this?

File: tests/utils/mock_server.py [hunk 2]
```diff
@@ -1363,8 +1387,7 @@ def graphql():
                 return json.dumps({"data": {"project": {"runQueues": []}}})
 
         if "query GetRunQueueItem" in body["query"]:
-            ctx["run_queue_item_check_count"] += 1
-            if ctx["run_queue_item_check_count"] > 1:
```
Comment (kptkin) [inline tests/utils/mock_server.py:1367]: do you need to remove this ctx?

File: tests/utils/mock_server.py (hunk 1 above)
Comment (KyleGoyette) [reply tests/utils/mock_server.py:1196]: Yes

File: tests/utils/mock_server.py [hunk 3]
```diff
@@ -1326,6 +1344,12 @@ def graphql():
                     art["artifactType"] = {"id": 5, "name": "job"}
                 return {"data": {"project": {"artifact": art}}}
         if "query ArtifactManifest(" in body["query"]:
+            # TODO: use emulator here
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply tests/utils/mock_server.py:1347]: Oh yeah, I did it!

File: tests/utils/mock_server.py (hunk 2 above)
Comment (KyleGoyette) [reply tests/utils/mock_server.py:1367]: Yeah, I've modified the associated test and made a more clear context to indicate how the mock server should respond to the query based on context

**Review by kptkin — APPROVED** [review]
Awesome work digging into the artifact emulator 🔥

**Pull request merged**